app.rb

```diff
@@ -0,0 +1,199 @@
+    gets.chomp
+    puts ''
+    run
+  end
+
+  # define list_books method
+  def list_books
+    if @book.empty?
+      puts 'No books'
+      show_menu
+    end
+    puts 'List all books:'
+    puts ''
+    @book.each_with_index do |item, idx|
+      puts "#{idx} Title: #{item.title}, Author: #{item.author}"
+    end
+    show_menu
+  end
+
+  # define list_people method
+  def list_people
+    if @people.empty?
+      puts 'No people'
+      show_menu
+    end
+    puts 'List of people:'
+    puts ''
+    @people.each_with_index do |item, idx|
+      puts "#{idx} -- [#{item.class}] Name: #{item.name}, Age: #{item.age}"
+    end
+    show_menu
+  end
+
+  # define create_person method
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? Please input number: '
+    option = gets.chomp.to_i
+    case option
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Please select 1 or 2'
+    end
+  end
+
+  # define create_book method
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @book.push(book)
+    puts 'book successfully created!'
+    show_menu
+  end
+
+  # define create_rental method
+  def create_rental
+    puts 'Select a book from list [Enter number]: '
+    @book.each_with_index { |item, idx| puts "#{idx}-- #{item.title}, #{item.author}" }
+    index = gets.chomp.to_i
+    book = @book[index]
+    puts 'Select a person from list [Enter number]: '
+    @people.each_with_index { |item, idx| puts "#{idx}- #{item.name}" }
+    person_id = gets.chomp.to_i
+    person = @people[person_id]
+
+    print 'Date(yyyy/mm/dd): '
+    date = gets.chomp
+
+    rental = Rental.new(date, book, person)
+    puts rental
+    @rental.push(rental)
+
+    puts 'Rental successfully created!'
+
+    show_menu
+  end
+
+  # define list_rentals
+  def list_rentals
+    puts 'Select a person id (Enter person id)'
+    @people.each { |item| puts "id: #{item.id}, Person: #{item.name}" }
+    print 'person id: '
+    id = gets.chomp.to_i
+    @rental.each do |item|
+      puts "Date: #{item.date}, Book: #{item.book.title} by #{item.book.author}" if item.person.id.to_i == id
+    end
+    show_menu
+  end
+
+  private
+
+  # define create_student method
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]:'
+    puts ''
+
+    parent_permission = gets.chomp.upcase
+    new_student = Student.new(age, parent_permission, name)
+    @people.push(new_student)
+    puts 'Student created successfully'
+    show_menu
+  end
+
+  # define create_teacher method
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teachcer.new(age, specialization, name)
```

Comment:
- [ ] There is a tiny bug 🐛 when creating a `Teacher` as a result of the typo on this line. You have `Teachcer` instead of `Teacher`. Please fix it 👍

Reply:
@codepantha Thank you 😁